src/vs/workbench/parts/terminal/electron-browser/terminal.ts

```diff
@@ -65,15 +65,15 @@ export interface ITerminalService {
 
 	close(): TPromise<any>;
 	copySelection(): TPromise<any>;
-	createNew(): TPromise<any>;
-	focus(): TPromise<any>;
+	createNew(name?: string): TPromise<number>;
```

Comment:
As a recommendation I'd return the `ITerminalPanel` here and move `setActive`, `show`, `dispose` etc in it - the services seems to become quite large.